File: fe-ta/src/models/ProductColor.ts
import type { ProductMediaResponse } from "./ProductMedia";

/** Dùng khi thêm/sửa màu sản phẩm */
export interface ProductColorRequest {
  productId: number;
  colorName: string;       
  colorCode?: string;      
  isAvailable?: boolean;   
  note?: string;
}

/** Dùng khi lọc màu sản phẩm trong admin */
export interface ProductColorFilterRequest {
  productId: number;
  productName?: string;
  colorName?: string;
  colorCode?: string;
  isAvailable?: boolean;
  createdName?: string;
  updatedName?: string;
  fromUpdateTime?: string;
  toUpdateTime?: string;
  note?: string;
}

/** Dữ liệu trả về từ API — hiển thị ở UI */
export interface ProductColorResponse {
  id: number;
  productId: number;
  productName?: string;
  colorName: string;
  colorCode?: string;
  isAvailable: boolean;
  createUid?: number;
  writeIUid?: number;
  updateTime?: string;
  createdName?: string;
  updatedName?: string;
  note?: string;
  mediaList?: ProductMediaResponse[];
}

File: beSQLSugar/beSQLSugar/Infrastructure/Database/Enities/ProductColor.cs
using SqlSugar;

namespace beSQLSugar.Infrastructure.Database.Enities
{
    [SugarTable("product_color")]
    public class ProductColor
    {
        [SugarColumn(IsPrimaryKey = true, IsIdentity = true)]
        public int Id { get; set; }

        [SugarColumn(IsNullable = false)]
        public int ProductId { get; set; }

        [SugarColumn(ColumnDataType = "NVARCHAR(100)", IsNullable = false)]
        public string ColorName { get; set; } = string.Empty; // ví dụ: "Trắng Titan"

        [SugarColumn(ColumnDataType = "NVARCHAR(10)", IsNullable = true)]
        public string? ColorCode { get; set; } // ví dụ: "#FFFFFF"

        [SugarColumn(IsNullable = true)]
        public bool IsAvailable { get; set; } = true;
        [SugarColumn(IsNullable = true)]
        public int? CreateUid { get; set; }

        [SugarColumn(ColumnDataType = "NVARCHAR(100)", IsNullable = true)]
        public string? CreatedName { get; set; }

        [SugarColumn(IsNullable = true)]
        public int? WriteIUid { get; set; }

        [SugarColumn(ColumnDataType = "NVARCHAR(100)", IsNullable = true)]
        public string? UpdatedName { get; set; }

        [SugarColumn(IsNullable = true)]
        public DateTime? UpdateTime { get; set; }

        [SugarColumn(ColumnDataType = "NVARCHAR(500)", IsNullable = true)]
        public string? Note { get; set; }

        [Navigate(NavigateType.ManyToOne, nameof(ProductId))]
        public Product? Product { get; set; }

        [Navigate(NavigateType.OneToMany, nameof(ProductMedia.ColorId))]
        public List<ProductMedia>? MediaList { get; set; }
    }
}
